android/app/src/main/java/org/fossasia/openevent/viewmodels/MainActivityViewModel.java

```diff
@@ -0,0 +1,226 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.fossasia.openevent.OpenEventApp;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.Event;
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.SessionType;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Sponsor;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.data.extras.SocialLink;
+import org.fossasia.openevent.dbutils.LiveRealmDataObject;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.events.EventDownloadEvent;
+import org.fossasia.openevent.events.JsonReadEvent;
+import org.fossasia.openevent.events.MicrolocationDownloadEvent;
+import org.fossasia.openevent.events.RetrofitError;
+import org.fossasia.openevent.events.SessionDownloadEvent;
+import org.fossasia.openevent.events.SessionTypesDownloadEvent;
+import org.fossasia.openevent.events.SpeakerDownloadEvent;
+import org.fossasia.openevent.events.SponsorDownloadEvent;
+import org.fossasia.openevent.events.TracksDownloadEvent;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.DateConverter;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+import org.fossasia.openevent.utils.Utils;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import io.realm.Realm;
+import io.realm.RealmList;
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class MainActivityViewModel extends ViewModel {
+    private RealmDataRepository realmRepo;
+    private LiveData<Event> eventLiveData;
+    private Event event;
+    private boolean newInstance = false;
+    private MutableLiveData<String> facebookPageNameLiveData;
+    private CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    public MainActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Event> getEventSync() {
+        newInstance = false;
+        if (eventLiveData == null) {
+            event = realmRepo.getEventSync();
+            if (event == null) {
+                newInstance = true;
+                event = new Event();
```

Comment:
Why assigning a new Event here

Reply:
I have removed the null check from the MainActivity so have to pass a non null event always. Is there any alternative?

Reply:
No there is no alternative, the view model should check for null and handle it appropriately. If the event is null, simply don't push it in LiveData. If it is a special case and it is not working with RealmLiveData wrapper, use MutableLiveData instead. Also, if you are using getEventSync, there is no use of LiveDat IMO because the data will not change at all. 

The use case is far simpler than we have got here. Get the event/listen for the event, if it is valid and not null, pass to activity. If it is null or invalid, don't do anything. 

I think the problem is that you are adhering to the architecture and logic which was previously used to a point where the benefit of using this new architecture is turning out to be obsolete. When we change from a simple MVC pattern to a more complex MVVM or even more verbose MVP, we have to completely change the way we think data flow happens and how we interact with data. This leaves us with tonnes of room for innovation and improvisation. Don't think of just copying and pasting chunks of data from views into ViewModels and get convert it into working LiveData. 

The key is to analyze what operations we need from view and for what `events` we listen in view and create our ViewModel according to that

We also use these patterns to abstract away from using implementation specific details, i.e., Android so that it will be easy to unit test. If you are serious about contributing to OSS and/or this or any project for the long term, I highly recommend learning some of the unit testing. Not a requirement per this issue, but in general it will help you a lot in visualizing what an end product should look like not only in terms of functionality but also in terms of maintainability and customisation.